trace/HTTP.md

```diff
@@ -0,0 +1,114 @@
+# HTTP
+
+This document explains tracing of HTTP requests with OpenCensus.
+
+## Spans
+
+Implementations MUST create a span for outgoing requests at the
+client and a span for incoming requests at the server.
+
+Span name is formatted as:
+
+* /$path for outgoing requests.
+* /($path|$route) for incoming requests.
+
+If route cannot be determined, path is used to name the
+the span for outgoing requests.
+
+Port MUST be omitted if it is 80 or 443.
+
+Examples of span names:
+
+* /users
+* /messages/[:id]
+* /users/25f4c31d
+
+Outgoing requests should be a span kind of CLIENT and
+incoming requests should be a span kind of SERVER.
+
+## Propagation
+
+Propagation is how SpanContext is transmitted on the wire
+in an HTTP request.
+
+Implementations MUST allow users to set their own propagation
+format and MUST provide an implementation for B3 at least.
+
+If user doesn't set any propagation methods explicitly, B3 is used.
+
+The propagation method SHOULD modify a request object to insert
+a SpanContext or SHOULD be able to extract a SpanContext from a
+request object.
+
+## Status
+
+Implementations MUST set status if HTTP request or response
+is not successful (e.g. not 2xx). In redirection case, if
+the client doesn't have autoredirection support, request
+should be considered successful.
+
+Set status code to UNKNOWN (2) if the reason cannot be inferred
+at the callsite or from the HTTP status code.
+
+Don't set the status message if the reason can be inferred at
+the callsite of from the HTTP status code.
+
+## Message events
+
+In the lifetime of an incoming and outgoing request, the following
```

Comment:
this is easier to do from low-level libraries. should we mention explicitly chunked encoding or if it is satisfactory to use Content-Length?

Reply:
I think we need to define behavior for chunked and multipart requests. I will file an issue and follow-up with a proposal.